test/test_query.cpp

```diff
@@ -1956,6 +1956,108 @@ TEST(Query_size)
     CHECK_EQUAL(6, tv.size());
 }
 
+
+TEST(Query_SubtableExpression)
+{
+    Group g;
+
+    TableRef table = g.add_table("foo");
+
+    DescriptorRef subdescr;
+    table->add_column(type_Table, "integers", &subdescr);
+    subdescr->add_column(type_Int, "list");
+    table->add_column(type_Table, "strings", &subdescr);
+    subdescr->add_column(type_String, "list", nullptr, true);
+
+    table->add_empty_row(4);
+
+    auto set_int_list = [](TableRef subtable, const std::vector<int64_t>& value_list) {
+        size_t sz = value_list.size();
+        subtable->clear();
+        if (sz) {
+            subtable->add_empty_row(sz);
+            for (size_t i = 0; i < sz; i++) {
+                subtable->set_int(0, i, value_list[i]);
+            }
+        }
+    };
+    auto set_string_list = [](TableRef subtable, const std::vector<int64_t>& value_list) {
+        size_t sz = value_list.size();
+        subtable->clear();
+        subtable->add_empty_row(sz);
+        for (size_t i = 0; i < sz; i++) {
+            if (value_list[i] < 100) {
+                std::string str("Str_");
+                str += std::to_string(value_list[i]);
```

Comment:
Its a pain, but CI fails because our crusty android compiler doesn't have `std::to_string` support. But you can use `realm::util::to_string()` instead.